File: sugarchainmininglibrary/src/main/java/com/nugetzrul3/sugarchainmininglibrary/SugarMiner.java
package com.nugetzrul3.sugarchainmininglibrary;

import android.os.Bundle;
import android.os.Handler;
import android.os.Message;

public class SugarMiner {
    private static final String TAG = "SugarMiner";

    static {
        System.loadLibrary("sugarminer");
    }

    public enum Algorithms {
        YESPOWER,
        YESPOWERSUGAR,
        YESPOWERLITB,
        YESPOWERIOTS,
        YESPOWERMBC,
        YESPOWERITC,
        YESPOWERISO,
    }

    private static Handler sHandler;

    public SugarMiner(Handler handler) {
        sHandler = handler;
    }
    public int beginMiner(String pool, String username, String pwd, int threads, Algorithms algorithm) {
        switch (algorithm) {
            case YESPOWER:
                return startMining(pool, username, pwd, threads, 0);
            case YESPOWERSUGAR:
                return startMining(pool, username, pwd, threads, 1);
            /*case YESPOWERLITB:
                return startMining(pool, username, pwd, threads, 3);
            case YESPOWERIOTS:
                return startMining(pool, username, pwd, threads, 4);
            case YESPOWERMBC:
                return startMining(pool, username, pwd, threads, 5);
            case YESPOWERITC:
                return startMining(pool, username, pwd, threads, 6);
            case YESPOWERISO:
                return startMining(pool, username, pwd, threads, 7);*/
            default:
                return -1;
        }
    }

    private static void output(String message) {
        Message msg = new Message();
        Bundle bundle = new Bundle();
        bundle.putString("log", message);
        msg.setData(bundle);

        if (sHandler != null) {
            sHandler.sendMessage(msg);
        }
    }

    public native int stopMining();
    private native int startMining(String url, String user, String password, int n_threads, int algo);
    public native int initMining();
}

File: sugarchainmininglibrary/src/main/java/com/nugetzrul3/sugarchainmininglibrary/SugarMiner.kt
package com.nugetzrul3.sugarchainmininglibrary

import android.os.Handler
import android.os.Bundle
import android.os.Message

class SugarMiner {
    companion object {
        private const val TAG: String = "SugarMiner"
        private var stringhandler: Handler? = null
        fun output(message: String) {
            val msg = Message()
            val bundle = Bundle()
            bundle.putString("log", message)
            msg.data = bundle
            if (stringhandler != null) {
                stringhandler!!.sendMessage(msg)
            }
        }

        init {
            System.loadLibrary("HelloWorld")
        }
    }

    enum class Algorithm {
        YESPOWER,
        YESPOWERSUGAR,
        YESPOWERLITB,
        YESPOWERIOTS,
        YESOPOWERMBC,
        YESPOWERITC,
        YESPOWERISO,
    }

    constructor() {}
    constructor(handler: Handler?) {
        stringhandler = handler
    }

    fun beginMiner(pool: String, username: String, pwd: String, threads: Int, algorithm: Algorithm?): Int {
        return when (algorithm) {
            Algorithm.YESPOWER -> startMining(pool, username, pwd, threads, 0)
            Algorithm.YESPOWERSUGAR -> startMining(pool, username, pwd, threads, 1)
            Algorithm.YESPOWERLITB -> startMining(pool, username, pwd, threads, 2)
            Algorithm.YESPOWERIOTS -> startMining(pool, username, pwd, threads, 3)
            Algorithm.YESOPOWERMBC -> startMining(pool, username, pwd, threads, 4)
            Algorithm.YESPOWERITC -> startMining(pool, username, pwd, threads, 5)
            Algorithm.YESPOWERISO -> startMining(pool, username, pwd, threads, 6)
            else -> -1
        }
    }


    private external fun startMining(pool: String, username: String, pwd: String, threads: Int, algorithm: Int): Int
    external fun stopMining(): Int

}
